coderd/coderd.go

```diff
@@ -823,6 +823,7 @@ func New(options *Options) *API {
 					r.Post("/convert-login", api.postConvertLoginType)
 					r.Delete("/", api.deleteUser)
 					r.Get("/", api.userByName)
+					r.Get("/parameters", api.userParameters)
```

Comment:
If we need to do an extra API call, I would make it more descriptive, for instance:

`/users/me/auto-fill/build-parameters`.

We could leave and use auto-fill for future purposes too.

Reply:
I should explain this in the code: the reason I went with the more generic name was for forward compatibility with “user_paramater” as a standalone concept (e.g key value mapping in user settings) as requested by a few users.

Reply:
I hear you. User parameters just look too generic to me. I'd expose this as the auto-fill endpoint in case we want to reuse it for other purposes. Of course, it is opinionated.